File: task2/helloworld.ex
defmodule PrimeNumbers do
  def is_prime(num) do
    if num < 2 do
      false
    else
      Enum.all?(2..round(:math.sqrt(num)), fn i -> rem(num, i) != 0 end)
    end
  end

  def print_primes(n) do
    IO.puts("Enter a number: ")
    input = IO.gets("")

    n = String.trim(input) |> String.to_integer()

    IO.puts("Prime numbers up to #{n}:")
    Enum.each(2..n, fn i ->
      if is_prime(i) do
        IO.puts(i)
      end
    end)
  end
end

PrimeNumbers.print_primes(10)

File: task2/helloworld.c
#include <stdio.h>
#include <math.h>

int isPrime(int num) {
    if (num < 2) {
        return 0;
    }
    for (int i = 2; i <= sqrt(num); i++) {
        if (num % i == 0) {
            return 0;
        }
    }
    return 1;
}

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);

    printf("Prime numbers up to %d:\n", n);
    for (int i = 2; i <= n; i++) {
        if (isPrime(i)) {
            printf("%d\n", i);
        }
    }

    return 0;
}
